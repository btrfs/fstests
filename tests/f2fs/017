#! /bin/bash
# SPDX-License-Identifier: GPL-2.0
# Copyright (c) 2025 Chao Yu.  All Rights Reserved.
#
# FS QA Test No. f2fs/017
#
# This testcase tries to check stability of mount result
# w/ mount options for zoned device and their combination.
#
. ./common/preamble
_begin_fstest auto quick mount

_require_zoned_device "$TEST_DEV"
_test_unmount >> $seqres.full 2>&1

options=(
#	mount_option			mkfs_option
	"background_gc=on"		"blkzone"		\
	"background_gc=off"		"blkzone"		\
	"background_gc=sync"		"blkzone"		\
	"mode=adaptive"			"blkzone"		\
	"mode=lfs"			"blkzone"		\
	"mode=fragment:segment"		"blkzone"		\
	"mode=fragment:block"		"blkzone"		\
	"discard_unit=block"		"blkzone"		\
	"discard_unit=segment"		"blkzone"		\
	"discard_unit=section"		"blkzone"		\
)

for ((i=0;i<${#options[@]};i=i+2))
do
	echo "Option#$i: ${options[$i]} : ${options[$((i+1))]}"
	_test_mkfs "-m" >> $seqres.full || _fail "mkfs failed"
	_test_mount "-o ${options[$i]}" >> $seqres.full 2>&1
	echo $?
	_test_unmount >> $seqres.full 2>&1
done

status=0
exit
