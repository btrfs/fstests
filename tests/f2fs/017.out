QA output created by 017
Option#0: background_gc=on : blkzone
0
Option#2: background_gc=off : blkzone
32
Option#4: background_gc=sync : blkzone
0
Option#6: mode=adaptive : blkzone
32
Option#8: mode=lfs : blkzone
0
Option#10: mode=fragment:segment : blkzone
32
Option#12: mode=fragment:block : blkzone
32
Option#14: discard_unit=block : blkzone
0
Option#16: discard_unit=segment : blkzone
0
Option#18: discard_unit=section : blkzone
0
